src/api/js/src/high-level/types.ts

```diff
@@ -109,98 +136,148 @@ export interface Context<Name extends string = 'main'> {
   ///////////////
   /** @category Functions */
   interrupt(): void;
+
   /** @category Functions */
   isModel(obj: unknown): obj is Model<Name>;
+
   /** @category Functions */
   isAst(obj: unknown): obj is Ast<Name>;
+
   /** @category Functions */
   isSort(obj: unknown): obj is Sort<Name>;
+
   /** @category Functions */
   isFuncDecl(obj: unknown): obj is FuncDecl<Name>;
+
   /** @category Functions */
   isApp(obj: unknown): boolean;
+
   /** @category Functions */
   isConst(obj: unknown): boolean;
+
   /** @category Functions */
   isExpr(obj: unknown): obj is Expr<Name>;
+
   /** @category Functions */
   isVar(obj: unknown): boolean;
+
   /** @category Functions */
   isAppOf(obj: unknown, kind: Z3_decl_kind): boolean;
+
   /** @category Functions */
   isBool(obj: unknown): obj is Bool<Name>;
+
   /** @category Functions */
   isTrue(obj: unknown): boolean;
+
   /** @category Functions */
   isFalse(obj: unknown): boolean;
+
   /** @category Functions */
   isAnd(obj: unknown): boolean;
+
   /** @category Functions */
   isOr(obj: unknown): boolean;
+
   /** @category Functions */
   isImplies(obj: unknown): boolean;
+
   /** @category Functions */
   isNot(obj: unknown): boolean;
+
   /** @category Functions */
   isEq(obj: unknown): boolean;
+
   /** @category Functions */
   isDistinct(obj: unknown): boolean;
+
   /** @category Functions */
   isArith(obj: unknown): obj is Arith<Name>;
+
   /** @category Functions */
   isArithSort(obj: unknown): obj is ArithSort<Name>;
+
   /** @category Functions */
   isInt(obj: unknown): boolean;
+
   /** @category Functions */
   isIntVal(obj: unknown): obj is IntNum<Name>;
+
   /** @category Functions */
   isIntSort(obj: unknown): boolean;
+
   /** @category Functions */
   isReal(obj: unknown): boolean;
+
   /** @category Functions */
   isRealVal(obj: unknown): obj is RatNum<Name>;
+
   /** @category Functions */
   isRealSort(obj: unknown): boolean;
+
   /** @category Functions */
   isBitVecSort(obj: unknown): obj is BitVecSort<number, Name>;
+
   /** @category Functions */
   isBitVec(obj: unknown): obj is BitVec<number, Name>;
+
   /** @category Functions */
   isBitVecVal(obj: unknown): obj is BitVecNum<number, Name>;
+
+  /** @category Functions */
+  isArraySort(obj: unknown): obj is SMTArraySort<any, any, Name>;
```

Comment:
Is it necessary to use `any` here, or could it be `unknown` or `AnySort<Name>[]` or something? Using `any` tends to make things typecheck which really shouldn't typecheck.

Reply:
Nope, I changed the `any`s to be more specific now